batch/test/test_batch.py

```diff
@@ -674,6 +674,27 @@ def test_verify_no_access_to_metadata_server(client):
     assert "Connection timed out" in j.log()['main'], (str(j.log()['main']), status)
 
 
+def test_submit_batch_in_job(client):
```

Comment:
Is it possible to try and write a test that tries to submit a batch to the wrong batch instance and ensure it fails? Maybe build a separate client with a DeployConfig as an argument with the wrong setting.

Reply:
Do you mean to try to submit to a different namespace or submit through the internet instead of internal gateway? If the former, what namespace should we use when the tests run in default?

Reply:
I meant let's figure out what points of attack we are most worried about (eg. the ip tables rules get refactored and don't do what we want anymore).

We could try submitting to a different namespace. For example, if the job is running in a PR or dev namespace, we try submitting a batch to the default batch instance. This should fail. If the job is running in the default namespace, then we expect it should succeed or we just skip the test.

I don't like having namespaces be interdependent though. Another idea is to not expect any response at all for the packets you've blocked. Maybe add a timeout to the request which uses a simple ClientSession to make the request? We expect a timeout error. If we get a different kind of error 502, 504, or success then we should fail the test. I'm not sure how to test contacting an external VM. Maybe the behavior is the same?

I can try and think about this more later if these proposals aren't good enough.

Reply:
Ah I see. In this case to test the iptables rules we would have to try to contact a different internal ip that is not internal gateway, but I don't know of one that is guaranteed to exist. We could reserve an IP address in that subnet purely for testing purposes..

For the submitting to a different namespace, I suppose I can add the test that expects a different result whether it is running in a PR namespace vs default.